services/QuillLMS/spec/controllers/snapshots_controller_spec.rb

```diff
@@ -9,72 +9,88 @@
     allow(controller).to receive(:current_user).and_return(user)
   end
 
-  context "#count" do
+  context "#actions" do
     let(:cache_key) { 'CACHE_KEY' }
-    let(:query_name) { 'active-classrooms' }
     let(:timeframe_name) { 'last-30-days' }
-    let(:query_class) { controller.class::QUERIES[query_name] }
     let(:now) { DateTime.current }
     let(:current_snapshot_stub) { 'CURRENT' }
     let(:previous_snapshot_stub) { 'PREVIOUS' }
     let(:school) { create(:school) }
     let(:user) { create(:user, administered_schools: [school]) }
     let(:school_ids) { [school.id.to_s] }
+    let(:controller_actions) {
+      [
+        [:count, 'active-classrooms'],
+        [:top_x, 'most-active-schools']
+      ]
+    }
 
     before do
       allow(controller).to receive(:cache_key_for_timeframe).and_return(cache_key)
       allow(DateTime).to receive(:current).and_return(now)
     end
 
-    it 'should return the value in the cache if it is available' do
+    it 'should return the value in the cache if it is available for all actions' do
       payload = {
         "current" => "CURRENT",
         "previous" => "PREVIOUS"
       }
 
-      expect(Rails.cache).to receive(:read).with(cache_key).and_return(payload)
+      expect(Rails.cache).to receive(:read).exactly(controller_actions.length).times.with(cache_key).and_return(payload)
 
-      get :count, params: { query: query_name, timeframe: timeframe_name, school_ids: school_ids }
+      controller_actions.each do |action, query_name|
```

Comment:
Interesting I haven't seen this construction before.

Reply:
It does make me wonder if these should be the same endpoint but I think that would be premature.

Reply:
I'm a little uncertain about all of this.  My thought was:
1. The payload shapes returned by these two snapshot types is different, so I figured it made sense for them to be two different actions.
2. But other than response shape, they basically are the same, so I didn't want to duplicate a bunch of code.

I could easily be convinced to do something different here, but it was the awkward compromise I settled on.